### PR TITLE
feat: added typescript types to ClassPrefix component

### LINKED ISSUE #13547: Add TypeScript types to ClassPrefix

### PR DESCRIPTION
**Pull request by SunnyJohal**
Closes #13547 

added typescript types to ClassPrefix component

#### Changelog

**New**

- added typescript types to ClassPrefix component

**Changed**

- file extension from .js to .tsx

**Removed**

- n/a

#### Testing / Reviewing

- Run storybook and check that the components are rendering.
- Run unit tests.

### CODE REVIEW

**Review by tay1orjones — APPROVED** [review]

File: packages/react/src/components/ClassPrefix/index.tsx [hunk 1]
```diff
@@ -6,10 +6,21 @@
...
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { PropsWithChildren } from 'react';
 import { PrefixContext } from '../../internal/usePrefix';
 
-function ClassPrefix({ children, prefix }) {
+interface ClassPrefixProps {
+  /**
+   * @description The value used to prefix the CSS selectors
```
Comment (tay1orjones) [inline packages/react/src/components/ClassPrefix/index.tsx]: The `@description` JSDoc annotation is [not supported](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#unsupported-tags)

```suggestion
   * The value used to prefix the CSS selectors
```

Additional usages of `@description` in this PR will need updated as well

**Review by tay1orjones — CHANGES_REQUESTED** [review]
Sorry I just noticed this

**Comment by tw15egan** [comment]
@SunnyJohal the `@description` comments are still present

**Review by tw15egan — APPROVED** [review]

**Review by alisonjoseph — APPROVED** [review]

**Comment by alisonjoseph** [comment]
recheck

**Comment by SunnyJohal** [comment]
Hi I'm going to close this branch and create a new one as I have accidentally rebased prev changes (the issue was my earlier commit was from an e-mail address from a company I was working with, I don't have access to that e-mail anymore)

**Pull request closed without merging**